### PR TITLE
added flag for copy_md

### PR DESCRIPTION
**Pull request by Shellyber**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/25256

## Description
Added flag for copy md files.

## Must have
- [x] Tests

### CODE REVIEW

**Comment by Shellyber** [comment]
Example of a build with content-new.zip without the md files.
https://app.circleci.com/pipelines/github/demisto/content/15037/workflows/ee38d4d0-2874-4185-adc4-70cea2c57130/jobs/56862/artifacts

**Comment by Shellyber** [comment]
Working on a test

File: demisto_sdk/commands/create_artifacts/content_creator.py [hunk 1]
```diff
@@ -278,19 +278,21 @@ def copy_dir_md(self, dir_path, bundle):
 
             shutil.copyfile(path, os.path.join(bundle, new_path))
 
-    def copy_dir_files(self, *args):
+    def copy_dir_files(self, *args, copy_md=False):
```
Comment (Itay4) [inline demisto_sdk/commands/create_artifacts/content_creator.py:281]: `copy_md` will always be `False`
need to set it pass `True` where we do want the `md` files

Comment (Shellyber) [reply demisto_sdk/commands/create_artifacts/content_creator.py]: I've. changed the default to true cause maybe other functions uses this command. 
So I'll keep the flag as false wherever needed

**Comment by Shellyber** [comment]
Last build, all is working as expected
https://app.circleci.com/pipelines/github/demisto/content/15223/workflows/dfe535fc-8b9d-4493-8dd7-0ce912b9bda2/jobs/57138/steps

**Comment by Shellyber** [comment]
@Itay4 Waiting for you review

File: demisto_sdk/commands/create_artifacts/content_creator.py [hunk 2]
```diff
@@ -341,7 +345,7 @@ def copy_packs_content_to_old_bundles(self, packs):
                 if dir_name == 'TestPlaybooks':
                     self.copy_test_files(sub_dir_path)
                 elif dir_name == RELEASE_NOTES_DIR:
-                    continue
+                    self.copy_test_files(sub_dir_path)
```
Comment (Itay4) [inline demisto_sdk/commands/create_artifacts/content_creator.py:348]: why is this needed?

File: demisto_sdk/commands/create_artifacts/tests/content_creator_test.py [hunk 3]
```diff
@@ -185,3 +187,28 @@ def test_copy_file_to_artifacts(self):
...
         assert filecmp.cmp(file_path, os.path.join(self.content_repo, filename))
+
+    def test_content_legacy_with_no_md(self):
+        """
+        Given
+        - valid content dir
+        When
+        - copying the content folder to a content bundle(flatten files)
+        Then
+        Ensure no md files were copied to content bundle
```
Comment (Itay4) [inline demisto_sdk/commands/create_artifacts/tests/content_creator_test.py:198]:
```suggestion
        - Ensure no md files were copied to content bundle
```

File: demisto_sdk/commands/create_artifacts/tests/content_creator_test.py [hunk 4]
```diff
@@ -185,3 +187,28 @@ def test_copy_file_to_artifacts(self):
...
+        Ensure md files were copied to packs bundle.
```
Comment (Itay4) [inline demisto_sdk/commands/create_artifacts/tests/content_creator_test.py:199]:
```suggestion
```

File: demisto_sdk/tests/test_files/valid_content_repo/IndicatorFields/field.json [hunk 5]
```diff
@@ -0,0 +1,40 @@
+{
```
Comment (Itay4) [inline demisto_sdk/tests/test_files/valid_content_repo/IndicatorFields/field.json:1]: why do we need all these test files?

Comment (Shellyber) [reply demisto_sdk/tests/test_files/valid_content_repo/IndicatorFields/field.json:1]: Basically just checked that everything was created ok.
I can remove the json files.

File: demisto_sdk/commands/create_artifacts/tests/content_creator_test.py (hunk 4 above)
Comment (Shellyber) [reply demisto_sdk/commands/create_artifacts/tests/content_creator_test.py:199]: Why remove this?

Comment (Itay4) [reply demisto_sdk/commands/create_artifacts/tests/content_creator_test.py:199]: sorry, meant to add it as bullet: `- Ensure md files were copied to packs bundle.`

File: demisto_sdk/tests/test_files/valid_content_repo/IndicatorFields/field.json (hunk 5 above)
Comment (Itay4) [reply demisto_sdk/tests/test_files/valid_content_repo/IndicatorFields/field.json:1]: lets remove

File: demisto_sdk/commands/create_artifacts/content_creator.py (hunk 2 above)
Comment (Shellyber) [reply demisto_sdk/commands/create_artifacts/content_creator.py:348]: No need, forgot to remove.
Removed this now.

File: demisto_sdk/commands/create_artifacts/tests/content_creator_test.py [hunk 6]
```diff
@@ -185,3 +187,28 @@ def test_copy_file_to_artifacts(self):
...
+        bundle_packs = os.path.join(temp_bundles, 'bundle_packs')
+        bundle_content = os.path.join(temp_bundles, 'bundle_content')
+        content_creator = ContentCreator(artifacts_path=temp_bundles, content_version='2.5.0',
+                                         preserve_bundles=True, no_update_commonserver=True)
+        with ChangeCWD(self.valid_content_repo):
+            content_creator.create_content()
+
+            assert filecmp.cmp(f'{self.valid_content_repo}/Packs/FeedAzureValid/ReleaseNotes/1_1_1.md',
+                               f'{bundle_packs}/FeedAzureValid/ReleaseNotes/1_1_1.md')
+            assert f'{self.valid_content_repo}/Packs/FeedAzureValid/CHANGELOG.md' not in bundle_content
```
Comment (Itay4) [inline demisto_sdk/commands/create_artifacts/tests/content_creator_test.py:211]: the assertion here is wrong, `bundle_content` is just a path of the bundle
we need to check the file does not exist in the bundle

**Comment by Shellyber** [comment]
@Itay4 Please review

**Comment by Itay4** [comment]
@jochman please review changes in test utils as well

File: TestSuite/pack.py [hunk 7]
```diff
@@ -152,6 +156,19 @@ def create_json_based(
         obj.write_json(content)
         return obj
 
+    def create_text_based(
```
Comment (jochman) [inline TestSuite/pack.py:159]: can be private?

File: TestSuite/pack.py [hunk 8]
```diff
@@ -186,9 +203,14 @@ def create_incident_field(
             self,
             name,
             content: dict = None,
+            release_notes: str = False
```
Comment (jochman) [inline TestSuite/pack.py:206]: str = false?

Comment (Shellyber) [reply TestSuite/pack.py:206]: Sorry forgot to change that. doing it now

**Comment by Shellyber** [comment]
@jochman Fixed. Please review and approve!
